apps/api/src/services/agent-manager.service.ts

```diff
@@ -808,6 +814,100 @@ export class AgentManager {
     }
   }
 
+  /**
+   * Get competitions for all agents owned by a user
+   * @param userId User ID
+   * @param params Agent competitions parameters
+   * @returns Object containing competitions array and total count
+   */
+  async getCompetitionsForUserAgents(
+    userId: string,
+    params: AgentCompetitionsParams,
+  ) {
+    try {
+      console.log(
+        `[AgentManager] Retrieving competitions for user ${userId} agents with params:`,
+        params,
+      );
+
+      const {
+        data: pagingParams,
+        success,
+        error,
+      } = PagingParamsSchema.safeParse(params);
+      if (!success) {
+        throw new ApiError(500, `cannot parse paging: ${error}`);
+      }
+      // Get all agents owned by this user
+      const userAgents = await this.getAgentsByOwner(userId, pagingParams);
+      const agentIds = userAgents.map((agent) => agent.id);
+
+      if (agentIds.length === 0) {
+        console.log(`[AgentManager] User ${userId} has no agents`);
+        return { competitions: [], total: 0 };
+      }
+
+      // Get competitions for all user's agents
+      const results = await findUserAgentCompetitions(agentIds, params);
+
+      // Group by competition to avoid duplicates and format the
+      // data to match what's expected by the client
+      const agentCompetitions = new Map();
+      results.competitions.forEach((data) => {
+        if (!data.competitions) return;
+        const comp =
+          agentCompetitions.get(data.competitions.id) || data.competitions;
+        const agent = data.agents ? this.sanitizeAgent(data.agents) : undefined;
+
+        if (!Array.isArray(comp.agents)) comp.agents = [];
+
+        if (
+          typeof agent?.id === "string" &&
+          !comp.agents.find((a: AgentPublic) => a.id === agent?.id)
+        ) {
+          comp.agents.push(agent);
+        }
+
+        agentCompetitions.set(data.competitions.id, comp);
+      });
+
+      // TODO: since rankings are not really done, need to make a bunch of db calls and get rankings
+      //  for every competition in this result set and make sure to append the agent's rank to the
+      //  response data.
+      await Promise.all(
```

Comment:
This is the place rank is calculated and attached to the response data.  It all seems a bit hacky since we are making potentially 10+ database requests (one for each competition), but I don't know how else to do this right now.

Reply:
potentially related to #546